app/src/org/commcare/models/database/InterruptedFormState.java

```diff
@@ -33,12 +35,14 @@ public void readExternal(DataInputStream in, PrototypeFactory pf)
             throws IOException, DeserializationException {
         sessionStateDescriptorId = ExtUtil.readInt(in);
         formIndex = (FormIndex)ExtUtil.read(in, FormIndex.class, pf);
+        formRecordId = ExtUtil.readInt(in);
```

Comment:
this will fail when trying to recover a form which has been saved on older version of app as there will be nothing to read. Safety story on this PR should detail out the backward compatibility with this change and look into ways of having the old forms recover successfully after this is rolled out.